### ISSUE TITLE
Use standard promise

### ISSUE THREAD

**Issue by Kikobeats** [matthistuff/buffer-node#1]
I don't know if currently the documentations is wrong, but in the main example you are not handling the error under `.catch` method. It should be looks like:

```js
api.user.get()
  .then(function(response) {
    // handle success
    console.log(response)
  })
  .catch(function(err) {
    // handle error
    console.log(err)
  })
)
```

On the other hand, if you drop old node support you can use the native Promise library and drop the `promise` dependency

**Comment by matthistuff** [comment]
Good point. I would definitely drop support for older node versions and use the native Promise.

**Comment by marcsyp** [comment]
Any update on this?

**Comment by Kikobeats** [comment]
I think cath is working as expected; About removing promise, PR's are welcome 🙂
